addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -29,20 +29,22 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
             order_line = self.env['sale.order.line'].sudo().search([
                 ('id', 'in', self.order_line.ids),
                 ('event_booth_pending_ids', 'in', event_booth_pending_ids)])
-            booths = self.env['event.booth'].browse(event_booth_pending_ids)
-            new_registration_ids = [Command.create({
-                                        'event_booth_id': booth.id,
-                                        **kwargs.get('registration_values'),
-                                    }) for booth in booths]
-            if order_line:
-                event_booth_registration_ids = [Command.delete(reg.id)
-                                                for reg in order_line.event_booth_registration_ids] + new_registration_ids
-            else:
-                event_booth_registration_ids = new_registration_ids
+            booths = self.env['event.booth'].browse(event_booth_pending_ids).with_context(pricelist=self.pricelist_id.id)
+            if order_line.event_booth_pending_ids.ids != event_booth_pending_ids:
+                new_registrations_command = [Command.create({
+                                            'event_booth_id': booth.id,
+                                            **kwargs.get('registration_values'),
+                                        }) for booth in booths]
+                if order_line:
+                    event_booth_registrations_command = [Command.delete(reg.id)
+                                                    for reg in order_line.event_booth_registration_ids] + new_registrations_command
+                else:
+                    event_booth_registrations_command = new_registrations_command
+                values['event_booth_registration_ids'] = event_booth_registrations_command
 
             values.update(
                 event_id=booths.event_id.id,
-                event_booth_registration_ids=event_booth_registration_ids,
+                price_unit=sum(booth.booth_category_id.price_reduce for booth in booths),
```

Comment:
NB: in master (15.3), the price computation should only be managed in event_booth_sale ;).